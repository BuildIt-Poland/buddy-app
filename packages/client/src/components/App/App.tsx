import React from 'react';

import PageContainer from 'atoms/PageContainer';
import { BrowserRouter } from 'react-router-dom';
import { ROUTES } from 'shared/routes';
import { useAuth } from 'contexts/AuthContext';

const loadAuthenticatedApp = () => import('components/AuthenticatedApp');
const AuthenticatedApp = React.lazy(loadAuthenticatedApp);
const NotAuthenticatedApp = React.lazy(() =>
  import('components/NotAuthenticatedApp')
);

const App: React.FC = () => {
  const { isAuthenticated } = useAuth();

  React.useEffect(() => {
    loadAuthenticatedApp();
  }, []);

  return (
    <BrowserRouter basename={ROUTES.BASE}>
      <React.Suspense
        fallback={<PageContainer data-testid='loader' backGroundShape />}>
        {isAuthenticated ? <AuthenticatedApp /> : <NotAuthenticatedApp />}
      </React.Suspense>
    </BrowserRouter>
  );
};

export default App;
