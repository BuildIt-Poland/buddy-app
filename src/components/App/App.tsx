import React from 'react';
import ThemeProvider from '@material-ui/styles/ThemeProvider';
import CssBaseline from '@material-ui/core/CssBaseline';
import Typography from '@material-ui/core/Typography';

import theme from '../../styles/theme';

const App: React.FC = () => {
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <Typography>
        <span data-cy='root'>Buddy App</span>
      </Typography>
    </ThemeProvider>
  );
};

export default App;
