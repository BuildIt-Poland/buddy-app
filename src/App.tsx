import React, { useEffect } from 'react';
import { useLazyQuery, useMutation } from '@apollo/react-hooks';
import gql from 'graphql-tag';
import logo from './logo.svg';
import './App.css';

// TODO: move graphQL instances(queries, mutations) to a separate file
export const LOGIN_MUTATION = gql`
  mutation LoginMutation($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      token
    }
  }
`;

export const NEWBIES_QUERY = gql`
  query NewbiesQuery($filter: String) {
    newbies(filter: $filter) {
      id
      name
      email
    }
  }
`;

interface Newbie {
  id: string;
  name: string;
  email: string;
}

// TODO: create login mechanism
export const AUTH_TOKEN = 'auth_token';
const CREDS = { email: 'las12041991@gmail.com', password: '12345' };
const TOKEN = localStorage.getItem(AUTH_TOKEN);

const App: any = () => {
  const [getNewbies, { loading, error, data }] = useLazyQuery(NEWBIES_QUERY);
  const [login] = useMutation(LOGIN_MUTATION, {
    onCompleted: ({ login }) => localStorage.setItem(AUTH_TOKEN, login.token),
  });

  const newbies: [Newbie] = data && data.newbies;

  useEffect(() => {
    if (!TOKEN) {
      login({ variables: CREDS }).then(() =>
        getNewbies({ variables: { filter: 'test' } })
      );
    } else {
      getNewbies({ variables: { filter: 'test' } });
    }
  }, [login, getNewbies]);

  return (
    <div className='App'>
      <header className='App-header'>
        <img src={logo} className='App-logo' alt='logo' />
        <p>
          Edit <code>src/App.tsx</code> and save to reload.
        </p>
        <div>
          {error && `Error! ${error.message}`}
          {(loading || !newbies) && 'Loading...'}
          {newbies &&
            newbies.map(({ id, name, email }) => (
              <div key={id}>{`name: ${name}, email: ${email}`}</div>
            ))}
        </div>
        <a
          className='App-link'
          href='https://reactjs.org'
          target='_blank'
          rel='noopener noreferrer'>
          Learn React
        </a>
      </header>
    </div>
  );
};

export default App;
