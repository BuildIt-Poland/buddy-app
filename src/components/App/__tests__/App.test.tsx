import React from 'react';
import { create } from 'react-test-renderer';

import App from '../App';

jest.mock('@material-ui/core/CssBaseline', () => 'CssBaseline');
jest.mock('@material-ui/styles/ThemeProvider', () => 'ThemeProvider');
jest.mock('@material-ui/core/Typography', () => 'Typography');
jest.mock('../../../styles/theme.ts', () => {});
jest.mock('../../AppWrapper', () => 'AppWrapper');

describe('Component - App', () => {
  test('renders correctly', () => {
    const component = create(<App />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
