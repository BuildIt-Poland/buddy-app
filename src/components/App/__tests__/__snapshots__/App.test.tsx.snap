// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - App renders correctly 1`] = `
<ThemeProvider>
  <CssBaseline />
  <Typography>
    <span
      data-cy="root"
    >
      Buddy App
    </span>
  </Typography>
</ThemeProvider>
`;
